src/components/App.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from "react";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import getForecast from "../requests/getForecast";
+import SearchForm from "./SearchForm";
+import "../styles/App.css";
+
+function App() {
+  const [forecasts, setForecasts] = useState([]);
+  const [location, setLocation] = useState({ city: "", country: "" });
+  const [selectedDate, setSelectedDate] = useState(0);
+  const [searchText, setSearchText] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  useEffect(() => {
+    getForecast(setSelectedDate, setForecasts, setLocation);
+  }, []);
```

Comment:
This is causing a problem when the app first loads: No data is fetched when the app first mounts. Looking at the console, this is because this request is failing as getRequest does not know what the setErrorMessage function is. You could change this useEffect to:
```
  useEffect(() => {
    getForecast(
      setSelectedDate,
      setForecasts,
      setLocation,
      "",
      setErrorMessage
    );
  }, []);
```
so that it will now be called when the component loads. Note that the API provides a default of manchester if no value for a city is passed in, so we can just pass it the value of an empty string in this case as this useEffect will only be triggered when it first loads